isy-security/src/main/java/de/bund/bva/isyfact/security/oauth2/client/Authentifizierungsmanager.java

```diff
@@ -2,58 +2,120 @@
-     * will be updated.
+     * Attempts to authorize the client for the given {@code oauth2ClientRegistrationId} via its configured OAuth 2.0 Flow.
+     * After successful authentication the authenticated principal in the {@link SecurityContext} will be updated.
+     * <p>
      * The chosen OAuth 2.0 Flow will depend on the authorization grant type configured in the application properties.
      * The currently supported flows are:
      * <ul>
-     *     <li>Client Credentials Flow (grant type: client_credentials)</li>
-     *     <li>Resource Owner Password Credentials Flow (grant type: password)</li>
+     *     <li>Client Credentials (grant type: client_credentials)</li>
+     *     <li>Resource Owner Password Credentials (grant type: password)</li>
      * </ul>
+     * Additional properties for the Resource Owner Password Credentials Flow will be resolved via the matching
+     * registration in {@link IsyOAuth2ClientConfigurationProperties#registration}.
      *
-     * @param registrationId
-     *         ID of the OAuth 2.0 Client to authorize
+     * @param oauth2ClientRegistrationId
+     *         registration ID of the OAuth 2.0 Client to authorize
      * @throws AuthenticationException
      *         if authentication fails
      */
-    void authentifiziere(String registrationId) throws AuthenticationException;
+    void authentifiziere(String oauth2ClientRegistrationId) throws AuthenticationException;
 
     /**
      * Attempts to create and authorize a client with the given credentials via the OAuth 2.0 Client Credentials Flow.
-     * After successful authentication the authenticated principal in the {@link org.springframework.security.core.context.SecurityContext}
-     * will be updated.
+     * After successful authentication the authenticated principal in the {@link SecurityContext} will be updated.
      * <p>
-     * This method is only intended to ease migration to the Client Credentials Flow.
-     * If possible it is preferred to use {@link #authentifiziere(String)} instead.
+     * This method is only intended to allow OAuth 2.0 Client Authentication in cases where the credentials
+     * are obtained externally and cannot be configured in the application properties.
+     * If possible it is strongly preferred to use {@link #authentifiziere(String) authentication with a client registration ID} instead.
      *
+     * @param issuerLocation
+     *         Issuer used to query the discovery endpoints and set the token endpoint for authentication
      * @param clientId
      *         Client ID of the OAuth 2.0 Client to authorize
      * @param clientSecret
      *         Client secret of the OAuth 2.0 Client to authorize
+     * @throws AuthenticationException
+     *         if authentication fails
+     * @see #authentifiziereClient(String, String, String, String)
+     */
+    void authentifiziereClient(String issuerLocation, String clientId, String clientSecret) throws AuthenticationException;
```

Comment:
But it will not has any benefits from invokation point of view like from _service-gateway_ or _authentication-gateway_. 

Each external request will have in the HTTP-Header a client-certificate, which the gateway has to interpret and to read the optional attribut `BHKNZ` for a 2AF. So the gateway will not check, if the `BHKNZ` was set in the HTTP-Header, it will only set the read value of the `BHKNZ` as parameter in the _authentifiziereClient/System_ method.

To use this method without this optional `BHKNZ` parameter, the gateway needs to check first for an existing _BHKNZ value_ out of the HTTP-Header and has then to decide, which method it has to invoke. That makes no sense.

If the _client-credential_-flow in IAM will support a 2AF in the future, it will have no effect on the method-usage with the optional parameter `BHKNZ`, but the method without the `BHKNZ`-parameter can't be used any more.

Reply:
there will some application/systems, which will never work with the BHKNZ.
no changes